content/docs/guides/06_light_push_send_messages.md

```diff
@@ -76,22 +80,252 @@ The returned `Promise` will resolve once you are connected to a Waku peer.
+await waitForRemotePeer(node, protocols);
+```
+
+## Create an encoder for the message
+
+```ts
+const Encoder = createEncoder(CONTENT_TOPIC);
+```
+
+## Define the structure of the message
+
+```ts
+const ChatMessage = new protobuf.Type("ChatMessage")
+  .add(new protobuf.Field("timestamp", 1, "uint64"))
+  .add(new protobuf.Field("nick", 2, "string"))
+  .add(new protobuf.Field("text", 3, "bytes"));
+```
+
+## Create a message object and encode it 
+
+```ts
+const message = ChatMessage.create({
+  nick: "your nick",
+  timestamp: Date.now(),
+  text: utils.utf8ToBytes("your message"),
+});
+const protoMessage = ChatMessage.encode(message).finish();
+```
+
+## Push the encoded message to the network
+
+```ts
+await node.lightPush.push(Encoder, {
+  payload: protoMessage,
+});
+```
+
+# Receive messages
+
+To receive messages, you need to subscribe to the Waku Filter, decode the incoming payload using a decoder specific to the content topic of the message, and then pass the decoded message to a callback function for processing.
+
+## Import the required libraries
+
+```ts
+import * as utils from "@waku/byte-utils";
+import * as wakuCreate from "@waku/create";
+import { createDecoder, waitForRemotePeer } from "@waku/core";
+```
+
+## Define the Waku Node connection parameters
+
+```ts
+const MULTI_ADDR = "/dns4/node-01.ac-cn-hongkong-c.wakuv2.test.statusim.net/tcp/443/wss/p2p/16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm";
+const CONTENT_TOPIC = "/toy-chat/2/huilong/proto";
+const PROTOCOLS = ["filter"];
+```
+
+## Create the Waku Node
+
+```ts
+const node = await wakuCreate.createLightNode({ defaultBootstrap: true });
+```
+
+## Start the Waku Node
 
-If the dApp is not connected to any light push peer, an error is thrown.
+```ts
+await node.start();
+```
+
+## Wait for remote peers to be connected
+
+```ts
+await waitForRemotePeer(node, PROTOCOLS);
+```
+
+## Define the protobuf type for the messages
+
+```ts
+const ChatMessage = new protobuf.Type("ChatMessage")
+  .add(new protobuf.Field("timestamp", 1, "uint64"))
+  .add(new protobuf.Field("nick", 2, "string"))
+  .add(new protobuf.Field("text", 3, "bytes"));
+```
+
+## Create a decoder for the content topic
+
+```ts
+const Decoder = createDecoder(CONTENT_TOPIC);
+```
+
+## Subscribe to the messages
+
+```ts
+const unsubscribeFromMessages = await node.filter.subscribe([Decoder], (wakuMessage) => {
+  const messageObj = ChatMessage.decode(wakuMessage.payload);
+  onMessageReceived({
+    ...messageObj,
+    text: utils.bytesToUtf8(messageObj.text),
+  });
+});
+```
+
+## Process the received message by decoding it 
 
 ```ts
-import { WakuMessage } from "js-waku";
+const onMessageReceived = (messageObj) => {
+  // process the received message
+};
+```
 
-const wakuMessage = await WakuMessage.fromUtf8String(
-  "Here is a message",
-  `/light-push-guide/1/guide/proto`
-);
+## Unsubscribe from the messages, when you no longer need to receive them
+
+```ts
+await unsubscribeFromMessages();
```

Comment:
Here I would mention that `unsubscribe` is a function returned by `node.filter.subscribe` API